daemons/based/based_callbacks.c

```diff
@@ -1448,16 +1491,26 @@ cib_process_command(xmlNode *request, const cib_operation_t *operation,
     }
 
     if (!pcmk_is_set(call_options, cib_dryrun)
-        && pcmk__str_eq(section, XML_CIB_TAG_STATUS, pcmk__str_casei)) {
-        // Copying large CIBs accounts for a huge percentage of our CIB usage
+        && !pcmk__str_eq(op, PCMK__CIB_REQUEST_COMMIT_TRANSACT, pcmk__str_none)
```

Comment:
Do we need this if we're checking cib_transaction?

Reply:
This is mostly for future-proofing. `cib_dryrun` or `PCMK__CIB_REQUEST_COMMIT_TRANSACT` true guarantees that `cib_zero_copy` gets cleared. If those are both false, then either `cib_transaction` set or the status section as the target guarantees that `cib_zero_copy` gets set.

`cib_transaction` and `PCMK__CIB_REQUEST_COMMIT_TRANSACT` are mutually exclusive*. Currently there's no way the target for commit-transaction could be the status section either. So this condition is really saying "no matter what gets added here in the future, we always need to make a copy (i.e., clear `cib_zero_copy`) for commit-transaction." Otherwise, we'd be writing to the live CIB, and we couldn't abort and roll back.

*A commit-transaction request doesn't support being run within a transaction. I originally went with `cib_extend_transaction` for the call option name, as an attempt at clarity. `cib_in_transaction` might be a good middle ground if we choose to change it.

Reply:
On a related note, we should probably deprecate `cib_zero_copy` and keep the same logic internally -- maybe move it to `cib_perform_op()` if doing so doesn't break `cib_file`. Currently, if the caller sets `cib_zero_copy`, we completely ignore it already. It's either set or cleared in `cib_process_command()` by this if-else. It would be dangerous to allow a caller to make this decision.

Reply:
agreed